src/proxy.ts

```diff
@@ -0,0 +1,65 @@
+/*
+ * sonar-scanner-npm
+ * Copyright (C) 2022-2024 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+import { URL } from 'url';
```

Comment:
I see some changes in the logic for proxies, we can discuss this later

Reply:
Yes, the logic is now that the user provides proxy configuration through `sonar.scanner.proxy*`, and this works both in the bootstrapper (node) and scanner-engine/cli (java)

Reply:
As agreed, will add support for `HTTP(S)_PROXY` env variable